src/jit/lsra.cpp

```diff
@@ -635,30 +714,66 @@ LinearScan::associateRefPosWithInterval(RefPosition *rp)
     }
 }
 
-RefPosition *
-LinearScan::newRefPosition(
-    regNumber reg, LsraLocation theLocation,
-    RefType theRefType, GenTree * theTreeNode,
-    regMaskTP mask)
-{
-    RefPosition *newRP = newRefPositionRaw(theLocation, theTreeNode, theRefType);
+//---------------------------------------------------------------------------
+// newRefPosition: allocate and initialize a new RefPosition.
+//
+// Arguments:
+//     reg             -  reg number that identifies RegRecord to be associated 
+//                        with this RefPosition
+//     theLocation     -  LSRA location of RefPosition
+//     theRefType      -  RefPosition type
+//     theTreeNode     -  GenTree node for which this RefPosition is created
+//     mask            -  Set of valid registers for this RefPosition
+//     multiRegIdx     -  register position if this RefPosition corresponds to a
+//                        multi-reg call node.
+//
+// Return Value:
+//     a new RefPosition
+//                       
+RefPosition*
+LinearScan::newRefPosition(regNumber reg, 
+                           LsraLocation theLocation,
+                           RefType theRefType, 
+                           GenTree* theTreeNode,
+                           regMaskTP mask,
+                           unsigned multiRegIdx /* = 0 */)
+{
```

Comment:
I don't think we need the multiRegIdx for this constructor - in fact, I don't think we really need theTreeNode either. There is only one place we call it with a non-null tree node, and that is in `buildInternalRegisterUsesForNode' where it is not clear to me why we would ever encounter a PhysReg ref, since 'buildInternalRegisterDefsForNode` never creates one.